src/Raven.Server/Documents/Indexes/Index.cs

```diff
@@ -3813,24 +3813,24 @@ private void AssertQueryDoesNotContainFieldsThatAreNotIndexed(QueryMetadata meta
                         continue;
 #endif
 
-                    AssertKnownField(f, metadata.IndexName);
+                    AssertKnownField(f, metadata);
                 }
             }
         }
 
-        private void AssertKnownField(string f, string indexName)
+        private void AssertKnownField(string f, QueryMetadata queryMetadata)
         {
             // the catch all field name means that we have dynamic fields names
 
             if (Definition.HasDynamicFields || IndexPersistence.ContainsField(f))
                 return;
 
-            ThrowInvalidField(f, indexName);
+            ThrowInvalidField(f, queryMetadata);
         }
 
-        private static void ThrowInvalidField(string f, string indexName)
+        private static void ThrowInvalidField(string f, QueryMetadata queryMetadata)
         {
-            throw new ArgumentException($"The field '{f}' is not indexed for index '{indexName}', cannot query/sort on fields that are not indexed");
+            throw new ArgumentException($"The field '{f}' is not indexed, cannot query/sort on fields that are not indexed in query: " + queryMetadata.QueryText);
```

Comment:
Why did you change the index name to query text here? The index name was suitable I think

Reply:
Let's have both - index name (as it was) and the query.